File: 330final/highc.jl
#!/usr/bin/julia

#main method to run bulk of code
function main()
    #declare and instantiate variables
    minVal::Int64 = 0
    maxVal::Int64 = 0
    max::Int64 = 0
    #get start and end values
    print("Minimun Value: ")
    minVal = parse(Int64,readline())
    print("Maximum Value: ")
    maxVal = parse(Int64,readline())

    #main loop
    for j in minVal:maxVal
        i = j
        #loop to calculate largest number in collatz sequence
        while i != 1
            if i % 2 == 0   #if even divide by 2
                #check for max
                if i > max
                    max = i
                end
                i = i ÷ 2
            else            #if odd multiply by 3 and add 1
                #check for max
                if i > max
                    max = i
                end
                i = i * 3 + 1
            end
        end
    end
    #print max
    println(max)
end

main()

exit(0)

File: 330final/highc.go
package main

import "fmt"	//import format for I/O

func main(){
	//instantiate variables
	var i int
	var minVal int
	var maxVal int
	fmt.Print("Minimum Value: ")
	fmt.Scan(&minVal)
	fmt.Print("Maximum Value: ")
	fmt.Scan(&maxVal)
	max := 0

	//main loop
	for j := minVal; j <= maxVal; j++ {
		i = j	//save j value in i
		//loop to calcualte largest number in collatz sequences
		for ok := true; ok; ok = (i != 1){
			if i % 2 == 0 {	//if even divide by 2
				if i > max {
					max = i
				}
				i /= 2
			} else {	//if odd multiply by 3 and add 1
				if i > max {
					max = i
				}
				i = i * 3 + 1
			}
		}
		
	}
	fmt.Println(max)
}
